pkg/sql/opt_exec_factory.go

```diff
@@ -123,7 +123,7 @@ func (ef *execFactory) ConstructScan(
 		scan.lockingWaitPolicy = descpb.ToScanLockingWaitPolicy(params.Locking.WaitPolicy)
 	}
 	scan.localityOptimized = params.LocalityOptimized
-	if !ef.isExplain {
+	if !ef.isExplain && !isVerifyIndexCountQuery(ef.planner.stmt.SQL, int(tabDesc.GetID()), int(idx.GetID())) {
```

Comment:
would it be better to just exclude internal executor queries here? Any time we use an internal executor on a user table, it's because of a schema change. See `ef.planner.isInternalPlanner`

Reply:
Yea, I've changed this to exclude internal executor queries. Not necessary for index usage statistics. Thank you for looking into this.